prov/shm/src/smr_rma.c

```diff
@@ -157,8 +157,14 @@ ssize_t smr_generic_rma(struct smr_ep *ep, const struct iovec *iov,
 	total_len = ofi_total_iov_len(iov, iov_count);
 
 	smr_generic_format(cmd, peer_id, op, 0, data, op_flags);
-	if (total_len > SMR_INJECT_SIZE || op != ofi_op_write ||
-	    op_flags & FI_DELIVERY_COMPLETE) {
+	if (total_len <= SMR_MSG_DATA_LEN && op != ofi_op_read_req &&
+	    !(op_flags & FI_DELIVERY_COMPLETE)) {
+		smr_format_inline(cmd, iov, iov_count);
+	} else if (total_len <= SMR_INJECT_SIZE && op != ofi_op_read_req &&
+		   !(op_flags & FI_DELIVERY_COMPLETE)) {
```

Comment:
I understand and support the purpose of these changes but these checks are starting to get a little crazy. Can we simplify this in any way?

Reply:
I improved a little bit, but still not perfect.